### PR TITLE
refactor: remove function types

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes:

Removed all function types like:-
`VersionFunctionType`
`OperatorFunctionType`
`SemanticFunctionType`

## Context:

These types are causing issues in strictNullChecks in renovate.

### CODE REVIEW

**Comment by rarkins** [comment]
@RahulGautamSingh lots of conflicts here

**Comment by RahulGautamSingh** [comment]
> @RahulGautamSingh lots of conflicts here

Resolved all.

**Comment by RahulGautamSingh** [comment]
@rarkins @viceice Ready for review and merge.🟩

File: lib/shared.d.ts [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import type { SemVer } from "semver";
 export interface parsed {
```
Comment (viceice) [inline lib/shared.d.ts:1]: Can we rename this to `Pep440Version` in next PR?

**Review by viceice — APPROVED** [review]

File: lib/shared.d.ts (hunk 1 above)
Comment (RahulGautamSingh) [reply lib/shared.d.ts:1]: Took the name after the constant in parse function.
https://github.com/renovatebot/pep440/blob/eefd68be4df448ac6ce3ab370317e6e94df95ee9/lib/version.js#L54-L75
So we can change the name for sure.

**Review by viceice — APPROVED** [review]

**Pull request merged**